examples/data-objects/datastore-race/src/index.tsx

```diff
@@ -0,0 +1,252 @@
+            new Map(),
+        );
+        const container = await loader.createDetachedContainer(containerRuntime.codeDetails);
+        const res = await container.request({ url: "" });
+        return res.value as MergeableClicker;
+    },
+};
+
+export const NormalContainerFactory: IRuntimeFactory =  {
+    get IRuntimeFactory() {return this;},
+    instantiateRuntime: async (context: IContainerContext)=>{
+        const runtime: ContainerRuntime =  await ContainerRuntime.load(
+            context,
+            [["default", Promise.resolve(MerableClickerInstantiationFactory)]],
+            async (req)=>{
+                const rp = RequestParser.create(req);
+                const ds =  await runtime.getRootDataStore(rp.pathParts[0] ?? "default", true);
+                return ds.request(rp.pathParts.length > 1 ? rp.createSubRequest(1) : { url: "" });
+            },
+            {
+                enableSummarizerNode: false,
+                generateSummaries: false,
+            });
+
+        if (runtime.existing === false) {
+            await runtime.createRootDataStore("default", "default");
+        }
+
+        return runtime;
+    },
+};
+
+export const RaceContainerFactory: IRuntimeFactory = {
+    get IRuntimeFactory() {return this;},
+    instantiateRuntime: async (context: IContainerContext)=>{
+        let current: MergeableClicker;
+        let lastElm: HTMLElement;
+        const viewProxy: IFluidHTMLView = {
+            get IFluidHTMLView() {return this;},
+            render(elm: HTMLElement, options?: IFluidHTMLOptions) {
+                lastElm = elm;
+                current.render(elm);
+            },
+        };
+
+        const runtime =  await ContainerRuntime.load(
+            context,
+            [["default", Promise.resolve(MerableClickerInstantiationFactory)]],
+            async (req)=> {
+                return  { status: 200, mimeType: "fluid/object", value: viewProxy };
+            },
+            {
+                generateSummaries:false,
+                enableSummarizerNode:false,
+            });
+
+        const local = current = await MerableClickerInstantiationFactory.createLocal(runtime);
+
+        runtime.once("connected", ()=>{
+            runtime.once("dataStoreRaceResolved", (id, channel)=>{
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                channel.request({ url:"" }).then(async (req)=>{
+                    const realClicker = req.value as MergeableClicker;
+                    current = realClicker;
+                    ReactDOM.unmountComponentAtNode(lastElm);
+                    local.merge(realClicker);
+                    viewProxy.render(lastElm);
+                    local.dispose();
+                });
+            });
+
+            const snapshot = local.snapshot();
+            // eslint-disable-next-line no-null/no-null
+            runtime.raceDataStore("default", { entries: snapshot, id: null }, "default");
```

Comment:
we've got out local copy, and our resolution handle set up, so kick off the race